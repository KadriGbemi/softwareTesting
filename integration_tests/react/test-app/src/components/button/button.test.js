import ReactDOM from "react-dom";
import { getQueriesForElement } from "@testing-library/react";

import renderer from "react-test-renderer"

import Button from ".";

describe("Test my button component", () => {
  const render = (component) => {
    const div = document.createElement("div");
    ReactDOM.render(component, div);
    return getQueriesForElement(div);
  };
  it("Renders button correctly and does not crash", () => {
    const { getByTestId } = render(<Button label="Dont click me" />);
    expect(getByTestId("my-button")).toHaveTextContent("Dont click me");
  });
  it("renders save text", () => {
    const { getByTestId } = render(<Button label="save"></Button>);
    expect(getByTestId("my-button")).toHaveTextContent("save");
  });

  it("Matches snapshot", ()=>{
      const buttonSnapshot = renderer.create(<Button label="Click me now"/>)
      expect(buttonSnapshot).toMatchSnapshot()
  })
});
