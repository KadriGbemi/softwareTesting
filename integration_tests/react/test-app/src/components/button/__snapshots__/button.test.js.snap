// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test my button component Matches snapshot 1`] = `
<div
  className="button-style"
  data-testid="my-button"
>
  Click me now
</div>
`;
